presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -191,51 +201,66 @@ public GlueHiveMetastore(
     {
         this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
         this.hdfsContext = new HdfsContext(new ConnectorIdentity(DEFAULT_METASTORE_USER, Optional.empty(), Optional.empty()));
-        this.glueClient = createAsyncGlueClient(requireNonNull(glueConfig, "glueConfig is null"), stats.newRequestMetricsCollector());
+        this.glueClient = createAsyncGlueClient(requireNonNull(glueConfig, "glueConfig is null"), stats.newRequestMetricPublisher());
         this.defaultDir = glueConfig.getDefaultWarehouseDir();
         this.catalogId = glueConfig.getCatalogId().orElse(null);
         this.partitionSegments = glueConfig.getPartitionSegments();
         this.executor = requireNonNull(executor, "executor is null");
     }
 
-    private static AWSGlueAsync createAsyncGlueClient(GlueHiveMetastoreConfig config, RequestMetricCollector metricsCollector)
+    private static GlueAsyncClient createAsyncGlueClient(GlueHiveMetastoreConfig config, MetricPublisher metricPublisher)
     {
-        ClientConfiguration clientConfig = new ClientConfiguration()
-                .withMaxConnections(config.getMaxGlueConnections())
-                .withMaxErrorRetry(config.getMaxGlueErrorRetries());
-        AWSGlueAsyncClientBuilder asyncGlueClientBuilder = AWSGlueAsyncClientBuilder.standard()
-                .withMetricsCollector(metricsCollector)
-                .withClientConfiguration(clientConfig);
+        NettyNioAsyncHttpClient.Builder nettyBuilder = NettyNioAsyncHttpClient.builder()
```

Comment:
Have you done any perf testing on the NIO client ? This is a good starting point, but since this is an async IO client, I was curious if we saw any stalling or failures (e.g timeout issues) that we need to experiment with

Reply:
Thanks for calling this out. I haven't seen any timeout issues so far on the workloads I have run. I can share some comparison numbers between v1 and v2. I will also try another workload with slightly higher concurrency. 

Also, most glue calls in Presto are executed synchronously via our wrapper(that blocks on the future). Only high-fan-out calls like `batchCreatePartition`, `batchGetPartition`, etc, are true async. We had to use the v2 async client as we have been using Async API calls for such operations with v1 client as well.

Reply:
I ran some more tests with higher concurrency and didn't observe any issues with the current configs. We can probably continue to observe the behaviour over time and tweak the client configs. Based on the AWS docs, in most of the use-cases we won't need to tweak the default configs. Let me know what you think.